app/controllers/telegram_bot_controller.rb

```diff
@@ -0,0 +1,45 @@
+class TelegramBotController < Telegram::Bot::UpdatesController
+  include Telegram::Bot::UpdatesController::MessageContext
+  def start!(*)
+    respond_with :message, text: 'Hi. Entry your /email example@mail.com and then /password example'
+  end
+
+  def email!(email)
+    respond_with :message, text: 'Success'
+    session['email'] = email
+  end
+
+  def password!(password)
+    session['password'] = password
+    if user_valid?
+      respond_with :message, text: 'Success. Entry /channels to see all channels'
+    else
+      respond_with :message, text: 'Not valid email or password'
+    end
+  end
+
+  def channels!
+    if user_valid?
+      respond_with :message, text: channels.pluck(:name).join("\n")
+    else
+      respond_with :message, text: 'Not valid email or password'
+    end
+  end
+
+  private
+
+  def user_valid?
+    user = User.find_by(email: session['email'])
+    return if user.nil?
+    user.valid_password?(session['password'])
```

Comment:
`user_valid?` is not quite accurate here, perhaps something like `telegram_user?`

fyi: 
```
def telegram_user?
  telegram_user&.valid_password?(session[:password])
end

def telegram_user
    @telegram_user ||= User.find_by(session.only('email'))
end
```